os/kernel/debug/secure_log.c

```diff
@@ -0,0 +1,128 @@
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define SECLOG_ON       "1"
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private data
+ ****************************************************************************/
+static int security_level = 0;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: set_security_level
+ *
+ * Description:
+ *   This function read the level of security from secure storage and write it
+ *   in 'security_level' variable.
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+void set_security_level(void)
+{
+        sl_ctx hnd;
+        uint32_t ss_idx = 32;
+        hal_data output;
+
+        /* init */
+        if (SECLINK_OK != sl_init(&hnd)) {
+                printf("[set_security_level] Failed to sl_init\n");
+                return;
+        }
+
+        output.priv = NULL;
+        output.priv_len = 0;
+        output.data_len = 2;
+        output.data = (char *)malloc(output.data_len);
+
+        if (SECLINK_OK != sl_read_storage(hnd, ss_idx, &output)) {
+                printf("[set_security_level] Failed to sl_read_storage\n");
+                return;
+        }
+
+        if (!strncmp(output.data, SECLOG_ON, 2)) {
+                security_level = 1;
+        } else {
+                security_level = 0;
+        }
+
+        printf("set_security_level : %d\n", security_level);
+}
+
+/****************************************************************************
+ * Name: check_security_level
+ *
+ * Description:
+ *   This function returns the security_level.
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   0(disable) or 1(enable)
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+int check_security_level(void)
```

Comment:
It will be used to determine whether to print the assert log or not.
Before printing the assert log, we will check the security level with this function and decide whether to print or not.